internal/controller/ingress/tlsedge_controller.go

```diff
@@ -206,10 +224,20 @@ func (r *TLSEdgeReconciler) updateEdge(ctx context.Context, edge *ingressv1alpha
 }
 
 func (r *TLSEdgeReconciler) delete(ctx context.Context, edge *ingressv1alpha1.TLSEdge) error {
+	// Get all owned domains and delete them.
+	log := ctrl.LoggerFrom(ctx)
+
+	// The edge must be deleted before the domains can be deleted
+	log.Info("Deleting TLSEdge", "edge.ID", edge.Status.ID)
 	err := r.NgrokClientset.TLSEdges().Delete(ctx, edge.Status.ID)
 	if err == nil || ngrok.IsNotFound(err) {
 		edge.Status.ID = ""
+		log.Info("Deleted TLSEdge", "edge.ID", edge.Status.ID)
```

Comment:
Probably want to move this line up one so the message gets the ID before it is cleared out.

Reply:
Whoops! Thanks for the good set of eyes on that. I'll fix it